### PR TITLE
XW-3476: Fix usage of global state in Wikia Rail (release)

### PR DESCRIPTION
**Pull request by mszabo-wikia**
@Wikia/x-wing

### CODE REVIEW

File: skins/oasis/modules/BodyController.class.php [hunk 1]
```diff
@@ -206,7 +209,7 @@ public function getRailModuleList() {
 		}
 
 		// Content, category and forum namespaces.  FB:1280 Added file,video,mw,template
-		if ( $this->wg->Title->isSubpage() && $this->wg->Title->getNamespace() == NS_USER ||
+		if ( $title->isSubpage() && $title->getNamespace() == NS_USER ||
```
Comment (WikiaTech) [inline skins/oasis/modules/BodyController.class.php:212]: MAJOR Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1067)

File: skins/oasis/modules/BodyController.class.php [hunk 2]
```diff
@@ -141,13 +141,16 @@ public static function getUserPagesNamespaces() {
 	}
 
 	public function getRailModuleList() {
```
Comment (WikiaTech) [inline skins/oasis/modules/BodyController.class.php:143]: MAJOR The Cyclomatic Complexity of this function "getRailModuleList" is 40 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
MAJOR Reduce the number of returns of this function 6, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 64 issues
* ![MAJOR][MAJOR] 59 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WikiaDispatchableObject.class.php#L12](https://github.com/Wikia/app/blob/72713b5286d1a1310b6151c2e57d9b99e709fa49/includes/wikia/nirvana/WikiaDispatchableObject.class.php#L12): Class "WikiaDispatchableObject" has 33 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR][MAJOR] [WikiaDispatchableObject.class.php#L213](https://github.com/Wikia/app/blob/72713b5286d1a1310b6151c2e57d9b99e709fa49/includes/wikia/nirvana/WikiaDispatchableObject.class.php#L213): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L29](https://github.com/Wikia/app/blob/72713b5286d1a1310b6151c2e57d9b99e709fa49/skins/oasis/modules/BodyController.class.php#L29): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L30](https://github.com/Wikia/app/blob/72713b5286d1a1310b6151c2e57d9b99e709fa49/skins/oasis/modules/BodyController.class.php#L30): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L38](https://github.com/Wikia/app/blob/72713b5286d1a1310b6151c2e57d9b99e709fa49/skins/oasis/modules/BodyController.class.php#L38): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L38](https://github.com/Wikia/app/blob/72713b5286d1a1310b6151c2e57d9b99e709fa49/skins/oasis/modules/BodyController.class.php#L38): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L45](https://github.com/Wikia/app/blob/72713b5286d1a1310b6151c2e57d9b99e709fa49/skins/oasis/modules/BodyController.class.php#L45): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L51](https://github.com/Wikia/app/blob/72713b5286d1a1310b6151c2e57d9b99e709fa49/skins/oasis/modules/BodyController.class.php#L51): Reduce the number of returns of this function 6, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR][MAJOR] [BodyController.class.php#L67](https://github.com/Wikia/app/blob/72713b5286d1a1310b6151c2e57d9b99e709fa49/skins/oasis/modules/BodyController.class.php#L67): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L86](https://github.com/Wikia/app/blob/72713b5286d1a1310b6151c2e57d9b99e709fa49/skins/oasis/modules/BodyController.class.php#L86): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by rogatty — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 64 issues
* ![MAJOR][MAJOR] 59 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WikiaDispatchableObject.class.php#L12](https://github.com/Wikia/app/blob/021831801cbd0497e6736b0e6cffa3ad6936c352/includes/wikia/nirvana/WikiaDispatchableObject.class.php#L12): Class "WikiaDispatchableObject" has 33 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR][MAJOR] [WikiaDispatchableObject.class.php#L213](https://github.com/Wikia/app/blob/021831801cbd0497e6736b0e6cffa3ad6936c352/includes/wikia/nirvana/WikiaDispatchableObject.class.php#L213): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L29](https://github.com/Wikia/app/blob/021831801cbd0497e6736b0e6cffa3ad6936c352/skins/oasis/modules/BodyController.class.php#L29): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L30](https://github.com/Wikia/app/blob/021831801cbd0497e6736b0e6cffa3ad6936c352/skins/oasis/modules/BodyController.class.php#L30): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L38](https://github.com/Wikia/app/blob/021831801cbd0497e6736b0e6cffa3ad6936c352/skins/oasis/modules/BodyController.class.php#L38): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L38](https://github.com/Wikia/app/blob/021831801cbd0497e6736b0e6cffa3ad6936c352/skins/oasis/modules/BodyController.class.php#L38): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L45](https://github.com/Wikia/app/blob/021831801cbd0497e6736b0e6cffa3ad6936c352/skins/oasis/modules/BodyController.class.php#L45): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L51](https://github.com/Wikia/app/blob/021831801cbd0497e6736b0e6cffa3ad6936c352/skins/oasis/modules/BodyController.class.php#L51): Reduce the number of returns of this function 6, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR][MAJOR] [BodyController.class.php#L67](https://github.com/Wikia/app/blob/021831801cbd0497e6736b0e6cffa3ad6936c352/skins/oasis/modules/BodyController.class.php#L67): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [BodyController.class.php#L86](https://github.com/Wikia/app/blob/021831801cbd0497e6736b0e6cffa3ad6936c352/skins/oasis/modules/BodyController.class.php#L86): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by macbre — APPROVED** [review]

**Pull request merged**